src/categories/routes.ts

```diff
@@ -0,0 +1,85 @@
+import express from 'express'
+import { asyncWrap, validateRequest, raise404, raise400 } from '../utils'
+import { listCategories, createCategory, getCategory, modifyCategory, deleteCategory } from './service'
+import { categorySchemaValidator, updateCategorySchemaValidator } from './validations'
+import { getCategoryByName } from './models'
+
+const router = express.Router()
+
+router.get('/categories', asyncWrap(async (req, res) => {
+  const categories = await listCategories()
+  res.status(200).json(categories)
+}))
+
+router.get('/categories/:categoryId', asyncWrap(async (req, res) => {
+  const id = parseInt(req.params.categoryId)
```

Comment:
We should use the already installed express-validator framework for this kind of input validations or use regex on the route to ensure that the input is a string. If it isn't, the api should return 404 (since there are no endpoints like /categories/<string>

Reply:
Got it, I will update the PR with a validation with express-validator

Reply:
New validations on `params` have been added to the relevant endpoints to ensure that these lookups have a valid parameter that 1) exists (required) and 2) is an integer. Requests that don't follow these validations are rejected with a 400 Bad Request error.

Given that req.params is a string array though, it is still required to parse the parameter as an integer. However, I found out that parsing with "Number" is faster that parsing with "parseInt", so I changed that up as well.

Fixed with 24f9ea2dff10559dfcda7202610705a62e7430a1